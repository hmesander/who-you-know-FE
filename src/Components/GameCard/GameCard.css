.new-deck-container {
  width: 70%;
  margin: 2em auto;
}

.game-card-component {
  border-top: 1px solid #eee;
  background-color: #fff;
  box-shadow:
    0 -1px 1px rgba(0,0,0,0.15),
    0 -10px 0 -5px #eee,
    0 -10px 1px -4px rgba(0,0,0,0.15),
    0 -20px 0 -10px #eee,
    0 -20px 1px -9px rgba(0,0,0,0.15);
  width: 60%;
  min-height: fit-content;
  height: 60vh;
  margin: 2.5em auto;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  padding: 30px;
}

.game-image {
  margin-bottom: 1.5em;
}

.new-deck-button {
  width: 20%;
  height: 24px;
}

.guessfield-container {
  display: flex;
  flex-direction: column;
}

.submit-guess-button {
  height: 24px;
}

.next-card-button {
  font-family: Helvetica,Arial,sans-serif;
  background-color: #f6e312;
  border: 1px solid #e9ac1a;
  text-shadow: 0 1px 1px rgba(255,255,255,0.75);
  border-radius: 3px;
  color: #333;
  cursor: pointer;
  background-image: linear-gradient(to bottom, #f6e312 0%,#f9c80d 100%);
}

.guess-field {
  height: 20px;
  width: 50%;
  margin-right: 10px;
  border-width: 1px;
}

label {
  margin: 1.5em;
}

.feedback-container {
  height: 80px;
}

.results-card {
  border-top: 1px solid #eee;
  background-color: #fff;
  box-shadow: 5px 5px 5px rgba(0,0,0,0.15);
  width: 60%;
  min-height: fit-content;
  margin: 2.5em auto;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  padding: 30px;
}

.scoreboard-button-container {
  width: 50%;
  margin: 3em auto;
  display: flex;
  justify-content: space-evenly;
}

.review-button {
  height: 24px;
}