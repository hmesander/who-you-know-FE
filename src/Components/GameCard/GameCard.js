import React from 'react';
import PropTypes from 'prop-types';
import './GameCard.css';

const GameCard = (props) => {
  if (props.gameDeck && props.gameDeck.length) {
    return (
      <div className='game-card-component'>
        <div className='game-card-info-container'>
          <img src={props.gameDeck[0].image_url} alt='Profile' className='circle-image game-image'></img>
          <hr className='card-hr'></hr>
        </div>
        <div className='guessfield-container'>
          <label htmlFor='user-guess'>Name This Connection:</label>
          <span className='input-holder'>
            <input
              id='user-guess'
              className='guess-field input-fields'
              aria-label='Please Enter Your Guess'
              type='text'
              name='guess'
              value={props.guess}
              onChange={(event) => props.handleChange(event)}
            />
            <button className='submit-guess-button buttons' onClick={() => props.handleSubmit()}>Submit Guess</button>
          </span>
        </div>
        <div className='feedback-container'>
          <p>{props.feedback}</p>
          {props.feedback ? <button className='next-card-button' onClick={() => props.getNextCard()}>Next Card</button> : null}
        </div>
      </div>
    );
  } else if (props.gameOver) {
    let fraction = props.correct / props.total;
    let percentage = Math.round(fraction * 100);
    return (
      <div className='results-card'>
        <h3>Results:</h3>
        <p>Correct: {props.correct}</p>
        <p>Incorrect: {props.incorrect}</p>
        <p>Percentage: {percentage}%</p>
        <div className='scoreboard-button-container'>
          {props.incorrect ? <button className='review-button buttons' onClick={() => props.reviewIncorrect()}>Redo Incorrect</button> : null}
          <button className='get-deck-button buttons' onClick={() => props.getDeck()}>Get New Deck</button>
        </div>
      </div>
    );
  } else {
    return (
      <div className='new-deck-container'>
        <button className='new-deck-button buttons' onClick={() => props.getDeck()}>Get Deck</button>
      </div>
    );
  }
};

export default GameCard;

GameCard.propTypes = {
  gameDeck: PropTypes.array,
  getDeck: PropTypes.func,
  handleChange: PropTypes.func,
  handleSubmit: PropTypes.func,
  getNextCard: PropTypes.func,
  reviewIncorrect: PropTypes.func,
  guess: PropTypes.string,
  feedback: PropTypes.string,
  gameOver: PropTypes.bool,
  correct: PropTypes.number,
  total: PropTypes.number,
  incorrect: PropTypes.number
};