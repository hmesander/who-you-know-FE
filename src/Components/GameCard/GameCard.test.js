/*eslint-disable no-undef camelcase*/

import React from 'react';
import GameCard from './GameCard';
import { shallow } from 'enzyme';

describe('GAMECARD TESTS', () => {
  let wrapper;
  const mockSubmit = jest.fn();
  const mockNextCard = jest.fn();
  const mockChange = jest.fn();
  const mockDeck = [{ image_url: 'google.com' }];

  beforeEach(() => {
    wrapper = shallow(<GameCard handleSubmit={mockSubmit} getNextCard={mockNextCard}gameDeck={mockDeck} handleChange={mockChange}/>);
  });

  it('should match the snapshot when there is a gamedeck to play', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handle change when the user types in the input field', () => {
    const mockEvent = { target: { value: 'something' } };
    wrapper.find('.guess-field').simulate('change', mockEvent);

    expect(mockChange).toHaveBeenCalled();
  });

  it('should call handleSubmit when submit guess button is clicked', () => {
    const button = wrapper.find('.submit-guess-button');
    button.simulate('click');

    expect(mockSubmit).toHaveBeenCalled();
  });

  it('should call getNextCard when Next Card button is clicked', () => {
    wrapper = shallow(<GameCard feedback='correct' getNextCard={mockNextCard} gameDeck={mockDeck} />);
    const button = wrapper.find('.next-card-button');
    button.simulate('click');

    expect(mockNextCard).toHaveBeenCalled();
  });

  it('should match the snapshot when the game is over', () => {
    wrapper = shallow(<GameCard gameOver='true' />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call getDeck function when get deck button is clicked', () => {
    const mockGetDeck = jest.fn();
    wrapper = shallow(<GameCard gameOver='true' getDeck={mockGetDeck}/>);
    const button = wrapper.find('.get-deck-button');
    button.simulate('click');

    expect(mockGetDeck).toHaveBeenCalled();
  });

  it('should call getDeck function when new deck button is clicked', () => {
    const mockGetDeck = jest.fn();
    wrapper = shallow(<GameCard getDeck={mockGetDeck}/>);
    const button = wrapper.find('.new-deck-button');
    button.simulate('click');

    expect(mockGetDeck).toHaveBeenCalled();
  });

  it('should match the snapshot before the game', () => {
    wrapper = shallow(<GameCard />);

    expect(wrapper).toMatchSnapshot();
  });
});