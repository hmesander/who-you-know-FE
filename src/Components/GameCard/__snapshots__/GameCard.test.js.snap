// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GAMECARD TESTS should match the snapshot before the game 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GameCard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        className="new-deck-button"
        onClick={[Function]}
>
        Get Deck
</button>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Get Deck",
        "className": "new-deck-button",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Get Deck",
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          className="new-deck-button"
          onClick={[Function]}
>
          Get Deck
</button>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Get Deck",
          "className": "new-deck-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Get Deck",
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`GAMECARD TESTS should match the snapshot when the game is over 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GameCard
    gameOver="true"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Results:
</h3>,
        <p>
          Correct: 
</p>,
        <p>
          Incorrect: 
</p>,
        <button
          className="get-deck-button"
          onClick={[Function]}
>
          Get Deck
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Results:",
        },
        "ref": null,
        "rendered": "Results:",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Correct: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Correct: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Incorrect: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Incorrect: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Get Deck",
          "className": "get-deck-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Get Deck",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Results:
</h3>,
          <p>
            Correct: 
</p>,
          <p>
            Incorrect: 
</p>,
          <button
            className="get-deck-button"
            onClick={[Function]}
>
            Get Deck
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Results:",
          },
          "ref": null,
          "rendered": "Results:",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Correct: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Correct: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Incorrect: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Incorrect: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Get Deck",
            "className": "get-deck-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Get Deck",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`GAMECARD TESTS should match the snapshot when there is a gamedeck to play 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GameCard
    gameDeck={
        Array [
            Object {
              "image_url": "google.com",
            },
          ]
    }
    getNextCard={[Function]}
    handleSubmit={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="game-card-info-container"
>
          <img
                    alt="Profile Picture"
                    className="circle-image"
                    src="google.com"
          />
          <hr
                    className="card-hr"
          />
</div>,
        <div
          className="guessfield-container"
>
          <label
                    htmlFor="user-guess"
          >
                    Name This Connection:
          </label>
          <input
                    aria-label="Please Enter Your Guess"
                    className="guess-field input-fields"
                    id="user-guess"
                    name="guess"
                    onChange={[Function]}
                    type="text"
                    value={undefined}
          />
          <button
                    className="submit-guess-button"
                    onClick={[Function]}
          >
                    Submit Guess
          </button>
</div>,
        <div
          className="feedback-container"
>
          <p />
</div>,
      ],
      "className": "game-card-component",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="Profile Picture"
              className="circle-image"
              src="google.com"
/>,
            <hr
              className="card-hr"
/>,
          ],
          "className": "game-card-info-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Profile Picture",
              "className": "circle-image",
              "src": "google.com",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "card-hr",
            },
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              htmlFor="user-guess"
>
              Name This Connection:
</label>,
            <input
              aria-label="Please Enter Your Guess"
              className="guess-field input-fields"
              id="user-guess"
              name="guess"
              onChange={[Function]}
              type="text"
              value={undefined}
/>,
            <button
              className="submit-guess-button"
              onClick={[Function]}
>
              Submit Guess
</button>,
          ],
          "className": "guessfield-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Name This Connection:",
              "htmlFor": "user-guess",
            },
            "ref": null,
            "rendered": "Name This Connection:",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-label": "Please Enter Your Guess",
              "className": "guess-field input-fields",
              "id": "user-guess",
              "name": "guess",
              "onChange": [Function],
              "type": "text",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit Guess",
              "className": "submit-guess-button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Submit Guess",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p />,
            null,
          ],
          "className": "feedback-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          null,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="game-card-info-container"
>
            <img
                        alt="Profile Picture"
                        className="circle-image"
                        src="google.com"
            />
            <hr
                        className="card-hr"
            />
</div>,
          <div
            className="guessfield-container"
>
            <label
                        htmlFor="user-guess"
            >
                        Name This Connection:
            </label>
            <input
                        aria-label="Please Enter Your Guess"
                        className="guess-field input-fields"
                        id="user-guess"
                        name="guess"
                        onChange={[Function]}
                        type="text"
                        value={undefined}
            />
            <button
                        className="submit-guess-button"
                        onClick={[Function]}
            >
                        Submit Guess
            </button>
</div>,
          <div
            className="feedback-container"
>
            <p />
</div>,
        ],
        "className": "game-card-component",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="Profile Picture"
                className="circle-image"
                src="google.com"
/>,
              <hr
                className="card-hr"
/>,
            ],
            "className": "game-card-info-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Profile Picture",
                "className": "circle-image",
                "src": "google.com",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "card-hr",
              },
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                htmlFor="user-guess"
>
                Name This Connection:
</label>,
              <input
                aria-label="Please Enter Your Guess"
                className="guess-field input-fields"
                id="user-guess"
                name="guess"
                onChange={[Function]}
                type="text"
                value={undefined}
/>,
              <button
                className="submit-guess-button"
                onClick={[Function]}
>
                Submit Guess
</button>,
            ],
            "className": "guessfield-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Name This Connection:",
                "htmlFor": "user-guess",
              },
              "ref": null,
              "rendered": "Name This Connection:",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-label": "Please Enter Your Guess",
                "className": "guess-field input-fields",
                "id": "user-guess",
                "name": "guess",
                "onChange": [Function],
                "type": "text",
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit Guess",
                "className": "submit-guess-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Submit Guess",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p />,
              null,
            ],
            "className": "feedback-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            null,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
